### ISSUE TITLE
Create Smaller MBTiles

### ISSUE THREAD

**Issue by longlostbro** [osm2vectortiles/osm2vectortiles#455]
Is it possible to create vector mbtiles of only certain layers or remove layers from an existing mbtiles, excluding layers not wanted such as 'points of interest'.

I want the world in mbtiles but I only need landuse,water,roads so without the excess data it could be much smaller.

**Comment by ImreSamu** [comment]
I don't tested, but in theory -this quick & dirty solution - should work:    
* you have to remove the unwanted layers / columns from the [/osm2vectortiles.tm2source/data.yml] (https://github.com/osm2vectortiles/osm2vectortiles/blob/master/osm2vectortiles.tm2source/data.yml)
     (please test  :  `step by step - with small changes ` and with a `minimal data` ! )

this is not optimal, because you will load the all OpenStreetMap data to the PostgreSQL database! 

>  I only need landuse,water,roads 

you have to keep - this layers in the yml configuration file
```
Layer: 
  - id: landuse
 ...
```


Side notes:
*  probably  for the next version (V3)   - we need to create a "tutorial" or a  simple configuration/optimalisation"   for this use case 
*  **the data model ( the layers and the columns will be radically change  in the V3 )**

**Comment by longlostbro** [comment]
How do you create an mbtiles from a tm2 source?

**Comment by ImreSamu** [comment]
@longlostbro 

* **_"Be aware of legal issues with deploying OSM2VectorTiles tiles"_** #387


> How do you create an mbtiles from a tm2 source?

the curent toolchain (V2)  ~  valid only for this year,    probably the V3 will be a little different.

some related documentations ( valid only for the V2 ): 

*  planet : https://github.com/osm2vectortiles/osm2vectortiles/blob/master/USAGE.md
*  smaller extract :  http://osm2vectortiles.org/docs/own-vector-tiles/
*  see the `export` modul: https://github.com/osm2vectortiles/osm2vectortiles/blob/master/src/export/README.md

For background informations:
*  Check the [Thesis](https://cdn.rawgit.com/osm2vectortiles/bachelor-thesis/882a46977f7b984fc59be188335127d62921c44c/thesis.pdf)

_"The export component is responsible for rendering vector tiles using **osm2vectortiles.tm2source**
and the postgis component. Exports can be run together with a message queue like RabbitMQ
or standalone for smaller extracts where it is not necessary to divide the work into several
parts."_

Probably you have to check the [docker-compose.yml](https://github.com/osm2vectortiles/osm2vectortiles/blob/master/docker-compose.yml) configurations
to understand the **tm2source** configurations : https://github.com/osm2vectortiles/osm2vectortiles/search?l=YAML&q=tm2source&utf8=%E2%9C%93

if you want to render the full planet - [with a single renderer:    >= 45 days](https://github.com/osm2vectortiles/osm2vectortiles/issues/251#issuecomment-220775839) , so you have to setup a complex parallel render system.

**Comment by longlostbro** [comment]
I'm not sure I understood your reference to "Be aware of legal issues with deploying OSM2VectorTiles tiles" [#387](https://github.com/osm2vectortiles/osm2vectortiles/issues/387)
What I understand is it is claimed to be illegal to 

> to produce vector tiles which are reproducing their tile schema directly
to distribute and use such map tiles

If so, then what you are saying is that **although it is possible** to do as I request, and create a mbtiles with only the selective layers I require, producing a smaller file, **it is illegal**.

**Comment by ImreSamu** [comment]
@longlostbro 

I don't know your **use case** and technical background, and I am not a lawyer, ...

imho ( my personal opinion )
* If you want to use in production
   *  it is better to wait to [V3 schema that breaks compatibility with Mapbox Streets.](https://github.com/osm2vectortiles/osm2vectortiles/milestone/21)
       *  not only legal ( ""Be aware of legal issues ..."" )
       *  but technical reasons, 
             *  because if you find a "bug"/"problems" , probably we don't patch the V2.
* In the V3 - probably we have a new tm2source definitions ( with different layers ) 
     *   so you can practice -  "removing layers"- on the V2 , and you can learn some open source tools,   
           *   ( Maybe you can help us in the development or testing  )

**Comment by longlostbro** [comment]
Thank you for your responses. They have been very helpful. I see you are at 35% for the v3.0 milestone and have no due date. Is this something you expect to take weeks, months, or years? Just getting a ballpark estimate.

**Comment by lukasmartinelli** [comment]
> Thank you for your responses. They have been very helpful. I see you are at 35% for the v3.0 milestone and have no due date. Is this something you expect to take weeks, months, or years? Just getting a ballpark estimate.

Release is end of 2016.

**Comment by longlostbro** [comment]
Will v3 work with mapnik still?

**Comment by klokan** [comment]
Yes. We may need help of community on porting the existing open CartoCSS styles from MapBox schema to the new tile schema.

**Comment by longlostbro** [comment]
I assume the new schema will be under the MIT license. In late November I may be able to help out in that regard. When the time comes, where would I look to get started on the port?
